Various no-diff cleanups and debugging additions/fixes

BruceForstall opened the pull request:
1. Add debugging cVN/dVN dumpers of value numbers
2. Fix new loop debugging dumpers on x86
3. Display the bbID, not just bbNum, in more places
4. For ValueNum handle constants, display the handle constant type
5. Fix some grammar nits and try to rewrite some comments to read better

BruceForstall committed:
Various no-diff cleanups and debugging additions

1. Add debugging cVN/dVN dumpers of value numbers
2. Fix new loop debugging dumpers on x86
3. Display the bbID, not just bbNum, in more places
4. For ValueNum handle constants, display the handle constant type
5. Fix some grammar nits and try to rewrite some comments to read better

BruceForstall commented on src/coreclr/jit/valuenum.cpp:
```diff
@@ -4017,66 +4017,66 @@ ValueNum ValueNumStore::EvalCastForConstantArgs(var_types typ, VNFunc func, Valu
             }
             break;
         }
-            {
```
There's no code change here; just fixing the indenting by moving this open scope inside the 'case' below

BruceForstall commented on src/coreclr/jit/valuenum.cpp:
```diff
@@ -8972,7 +8976,7 @@ void ValueNumStore::vnDump(Compiler* comp, ValueNum vn, bool isPtr)
                 }
                 else
                 {
-                    printf("LngCns: ");
```
This is to be consistent with `IntCns`, above.

BruceForstall commented:
I extracted these out of a change with semantic changes to make that one more focused.

BruceForstall commented:
@AndyAyersMS PTAL
cc @dotnet/jit-contrib

AndyAyersMS commented on src/coreclr/jit/gentree.cpp:
```diff
@@ -10710,6 +10709,57 @@ void GenTree::SetIndirExceptionFlags(Compiler* comp)
+        case GTF_ICON_STATIC_ADDR_PTR:
+            return printf("GTF_ICON_STATIC_ADDR_PTR");
+        default:
+            assert(!"a forgotten handle flag");
```
Maybe not assert here? 

Slight preference for having methods like this just return strings, so they compose better into formatted output.

Also: we have both "Disp" and "dsp", should we settle on just one?

BruceForstall replied:
> Maybe not assert here?
> 
> Slight preference for having methods like this just return strings, so they compose better into formatted output.

Agreed, and fixed.

> Also: we have both "Disp" and "dsp", should we settle on just one?

Maybe? IMO, slight preference for "disp". Not sure it's worth the effort to change it.

AndyAyersMS reviewed (approved):
Feel free to ignore my feedback or address later.

BruceForstall committed:
Code review feedback

Have function `GenTree::gtGetHandleKindString` to return handle string;
let the caller print it.

BruceForstall commented:
CI seems really broken

BruceForstall commented:
/azp run runtime-coreclr superpmi-diffs, runtime-coreclr superpmi-replay, runtime
